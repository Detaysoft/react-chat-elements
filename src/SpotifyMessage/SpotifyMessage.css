.rce-mbox-spotify {
    margin-top: -2px;
    overflow: hidden;
    margin-right: -6px;
    margin-left: -6px;
    display: flex;
    border-radius: 5px;
}
