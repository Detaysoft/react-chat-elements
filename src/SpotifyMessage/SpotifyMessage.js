import React, { Component } from 'react';
import './SpotifyMessage.css';

const classNames = require('classnames');

export class SpotifyMessage extends Component {
    toUrl() {
        var formBody = [];
        var data = {
            uri: this.props.uri,
            theme: this.props.theme,
            view: this.props.view,
        };
        for (var property in data) {
            var encodedKey = encodeURIComponent(property);
            var encodedValue = encodeURIComponent(data[property]);
            formBody.push(encodedKey + "=" + encodedValue);
        }
        formBody = formBody.join("&");
        return formBody;
    }

    render() {
        if (!this.props.uri)
            return null;
        return (
            <div className="rce-mbox-spotify">
                <iframe
                    src={"https://open.spotify.com/embed?" + this.toUrl()}
                    width={this.props.width}
                    height={this.props.height}
                    frameBorder="0"
                    allowtransparency="true"></iframe>
            </div>
        );
    }
}

SpotifyMessage.defaultProps = {
    uri: '',
    theme: 'black',
    view: 'list',
    width: 300,
    height: 380,
}

export default SpotifyMessage;
