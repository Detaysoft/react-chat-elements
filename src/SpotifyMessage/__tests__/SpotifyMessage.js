import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SpotifyMessage from '../SpotifyMessage';

describe('SpotifyMessage component', () => {
  it('should render without issues', () => {
    const component = shallow(<SpotifyMessage />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
