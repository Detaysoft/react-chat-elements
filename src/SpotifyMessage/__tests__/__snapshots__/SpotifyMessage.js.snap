// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotifyMessage component should render without issues 1`] = `null`;
